assets/training/model_management/src/azureml/model/mgmt/utils/common_utils.py

```diff
@@ -221,3 +224,67 @@ def fetch_huggingface_model_info(model_id) -> ModelInfo:
         raise AzureMLException._with_error(
             AzureMLError.create(HuggingFaceErrorInFetchingModelInfo, model_id=model_id, error=e)
         )
+
+
+def get_system_time_utc():
+    """Return formatted system time in UTC."""
+    return "{0:%Y-%m-%d %H:%M:%S}".format(datetime.utcnow())
+
+
+def get_file_or_folder_size(path: Path, size: int = 0) -> int:
+    """Return file or folder size in bytes."""
+    if os.path.isfile(path):
+        return os.stat(path).st_size
+    for entry in os.scandir(path):
+        size += get_file_or_folder_size(entry)
+    return size
+
+
+def round_size(size: int) -> str:
+    """Round size."""
+    CONST = 1024
+    dim = ["B", "KB", "MB", "GB", "TB"]
+    count = 0
+    while size / CONST > 1:
+        count += 1
+        size /= CONST
+    return f"{size:.2f} {dim[count]}"
+
+
+def get_filesystem_available_space_in_kb(path: Path = "./"):
+    """Return filesystem available size in KB.
+
+    :return: Return available size on VM in KB
+    :rtype: int
+    """
+    partition = psutil.disk_usage(path)
+    available_size_bytes = partition.free
+    return available_size_bytes / 1024
+
+
+def get_git_lfs_blob_size_in_kb(git_dir: Path) -> int:
+    """Return total LFS blob size in KB.
+
+    :param git_dir: git directory containing LFS file pointers
+    :type git_dir: Path
+    :return: total LFS blob size in KB
+    :rtype: int
+    """
+    try:
+        # 1. Execute git lfs ls-files -s command to get size of all blobs recursively
+        # 2. Filter size using awk utility and translate them to KB.
+        #    Expectation is that each blob dimension would be in [KB, MB, GB]
+        # 3. Perform sum and return total size in KB
+        cmd = (
+            f"cd {str(git_dir)} && "
```

Comment:
Where are we stopping `git lfs` from cloning large files?

Reply:
here: 

![image](https://github.com/Azure/azureml-assets/assets/61145377/4c689059-6ff9-4a07-b1bf-5d18b6878a05)

Reply:
In `GITDownloader._download`